File: RESTful Samples/RESTfulEngine.PHPClient/AdoDataSource.php
<?php
/*
 * Copyright (c) 2015 by Windward Studios, Inc. All rights reserved.
 *
 * This software is the confidential and proprietary information of
 * Windward Studios ("Confidential Information").  You shall not
 * disclose such Confidential Information and shall use it only in
 * accordance with the terms of the license agreement you entered into
 * with Windward Studios, Inc.
 */

require_once("DataSource.php");

/** ADO data source implementation.
 */
class AdoDataSource extends DataSource {

	/** Creates a new instance of the data source. For $className and $connectionString values
	 * consult the ADO.NET data source provider being used.
	 */
	public static function newInstance($className, $connectionString)
	{
		return new AdoDataSource($className, $connectionString);
	}

	/** Returns an object representation of this data source suitable to be sent over to the server.
	 * Normally, this is used by Report classes and is useless for the client.
	 */
	public function getObject($name)
	{
		$data["Name"] = $name;
		$data["Type"] = "sql";
		$data["ClassName"] = $this->state["classname"];
		$data["ConnectionString"] = $this->state["connectionstring"];

		$this->applyVariables($data);

		return $data;
	}
	
	protected function __construct($className, $connectionString)
	{
		parent::__construct();

		$this->state["classname"] = $className;
		$this->state["connectionstring"] = $connectionString;
	}
}

?>

File: RESTful Samples/RESTfulEngine.CSharpClient/src/AdoDataSource.cs
/*
 * Copyright (c) 2015 by Windward Studios, Inc. All rights reserved.
 *
 * This software is the confidential and proprietary information of
 * Windward Studios ("Confidential Information").  You shall not
 * disclose such Confidential Information and shall use it only in
 * accordance with the terms of the license agreement you entered into
 * with Windward Studios, Inc.
 */

using System.Xml.Linq;

namespace RESTfulEngine.CSharpClient
{
    /// <summary>
    /// ADO data source provider.
    /// </summary>
    public class AdoDataSource : DataSource
    {
        private string className;
        private string connectionString;

        public AdoDataSource(string className, string connectionString)
        {
            this.className = className;
            this.connectionString = connectionString;
        }

        internal override XElement GetXml(string name)
        {
            var element = new XElement("Datasource",
                new XElement("Name", name),
                new XElement("Type", "sql"),
                new XElement("ClassName", className),
                new XElement("ConnectionString", connectionString)
                );
            if (Variables != null && Variables.Count > 0)
            {
                element.Add(GetVariablesXml());
            }
            return element;
        }
    }
}
